### PR TITLE
river-luatile: init at v0.1.1

### PR DESCRIPTION
**Pull request by pinpox**
###### Description of changes

Add river-luatile package

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/river-luatile/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  };
+
+  cargoSha256 = "sha256-udfsd1iONlDSQ/7mzzRNNhoJHmXJsxWdhqeKK/onx+4=";
+
+  buildInputs = [ luajit ];
+  nativeBuildInputs = [ pkg-config ];
+  PKG_CONFIG_PATH = "${openssl.dev}/lib/pkgconfig";
+
+  meta = with lib; {
+    homepage = "https://git.sr.ht/~novakane/rivercarro";
```
[the remark this answers is not in the document]
Comment (pinpox) [reply pkgs/applications/misc/river-luatile/default.nix]: Oops, sorry, you're totally right.

**Review by MaxVerevkin — COMMENTED** [review]
@pinpox I don't think I'm the right person to review this PR

File: pkgs/applications/misc/river-luatile/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    fetchSubmodules = true;
+    rev = "v${version}";
+    sha256 = "sha256-A8vx8jN4XUUI970ZsWLKBCd5lO9p3w63b9EiGwk/rCU=";
+  };
+
+  cargoSha256 = "sha256-udfsd1iONlDSQ/7mzzRNNhoJHmXJsxWdhqeKK/onx+4=";
+
+  buildInputs = [ luajit ];
+  nativeBuildInputs = [ pkg-config ];
+  PKG_CONFIG_PATH = "${openssl.dev}/lib/pkgconfig";
```
Comment (Kranzes) [inline pkgs/applications/misc/river-luatile/default.nix:25]: Why is this needed?

**Comment by Kranzes** [comment]
Here's an updated expression (mostly generated via [nix-init](https://github.com/nix-community/nix-init). I recommend using it instead and adjusting PR and Commit name to be for version 0.1.1 instead of 0.1.0.

```nix
{ lib
, rustPlatform
, fetchFromGitHub
, pkg-config
, luajit
}:

rustPlatform.buildRustPackage rec {
  pname = "river-luatile";
  version = "0.1.1";

  src = fetchFromGitHub {
    owner = "MaxVerevkin";
    repo = "river-luatile";
    rev = "v${version}";
    hash = "sha256-eZgoFbat7X/jh5udlNyIuTheBUCHpaVRbsojYLATO18=";
  };

  cargoHash = "sha256-Vqyt5bL1lVhy/Wxd+zF7Wugvb7dW1N9Kq2TTFSaodnE=";

  nativeBuildInputs = [
    pkg-config
  ];

  buildInputs = [
    luajit
  ];

  meta = with lib; {
    description = "Write your own river layout generator in lua";
    homepage = "https://github.com/MaxVerevkin/river-luatile";
    license = licenses.gpl3Only;
    platforms = platforms.linux;
    maintainers = with maintainers; [ pinpox ];
  };
}
```

**Comment by pinpox** [comment]
> Here's an updated expression (mostly generated via [nix-init](https://github.com/nix-community/nix-init). I recommend using it instead and adjusting PR and Commit name to be for version 0.1.1 instead of 0.1.0.

Must have been released just after me creating this PR. I took your new expression and updated the PR
